### PR TITLE
Update thinning.cpp

### PR DESCRIPTION
**Pull request by sturkmen72**
Update thinning.cpp

### CODE REVIEW

**Comment by sturkmen72** [comment]
@StevenPuttemans could you check this

File: modules/ximgproc/samples/thinning.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ using namespace cv;
 
 int main()
 {
-    Mat img = imread("opencv-logo.png", IMREAD_COLOR);
+    Mat img = imread("../data/opencv-logo.png", IMREAD_REDUCED_COLOR_2);
```
Comment (StevenPuttemans) [inline modules/ximgproc/samples/thinning.cpp]: I am not convinced that changes like this work. If the samples get build, the data will not be one folder up. It is only like this in the case of the source files. Thus I suggest to undo this change.

File: modules/ximgproc/samples/thinning.cpp [hunk 2]
```diff
@@ -19,8 +19,8 @@ int main()
 
     /// Apply thinning to get a skeleton
     Mat img_thinning_ZS, img_thinning_GH;
-    //ximgproc::thinning(img_binary, img_thinning_ZS, THINNING_ZHANGSUEN);
-    //ximgproc::thinning(img_binary, img_thinning_GH, THINNING_GUOHALL);
+    ximgproc::thinning(img_binary, img_thinning_ZS, ximgproc::THINNING_ZHANGSUEN);
+    ximgproc::thinning(img_binary, img_thinning_GH, ximgproc::THINNING_GUOHALL);
```
Comment (StevenPuttemans) [inline modules/ximgproc/samples/thinning.cpp]: This is wrong, the constants are not under te ximgproc namespace for the moment, and I am still awaiting response on why they are not. So this would break the sample.

Comment (StevenPuttemans) [inline modules/ximgproc/samples/thinning.cpp]: Forget this last remark, it should indeed be there or at least include `using namespace cv::ximgproc` in the beginning of the file!

File: modules/ximgproc/samples/thinning.cpp (hunk 1 above)
Comment (sturkmen72) [inline modules/ximgproc/samples/thinning.cpp]: hi @StevenPuttemans 
"../data/" used in many sample 
for example https://github.com/opencv/opencv_contrib/blob/master/modules/ximgproc/samples/disparity_filtering.cpp#L18-L20

indeed i don't know another way and i always move the opencv/samples/data folder to a proper place according to where the compiled exe files placed.

so still i see this change will be useful.

also IMREAD_REDUCED_COLOR_2 will load the image resized. it is also better to see result image.

Comment (StevenPuttemans) [inline modules/ximgproc/samples/thinning.cpp]: Well it is a matter of flavor. 
About the reduced flag, I would not do this, it is additional functionality that is NOT needed in this case and could confuse people.

**Comment by sovrasov** [comment]
:+1:
